crates/ibc/src/event.rs

```diff
@@ -140,63 +140,6 @@ pub struct IbcEvent {
     pub attributes: HashMap<String, String>,
 }
 
-fn validate_ibc_event_type(
-    namada_event: &Event,
-) -> Result<IbcEventType, EventError> {
-    if namada_event.kind().domain() != IbcEvent::DOMAIN {
-        return Err(EventError::InvalidEventType);
-    }
-
-    let event_type = namada_event.kind().sub_domain();
-
-    // TODO(namada#3229): validate IBC event types. eg:
-    //
-    // ```ignore
-    // if !matches!(
-    //    event_type,
-    //    "update_client" | "send_packet" | "write_acknowledgement"
-    // ) {
-    //     return Err(EventError::InvalidEventType);
-    // }
-    // ```
-
-    Ok(IbcEventType(event_type.to_owned()))
-}
-
-impl TryFrom<&Event> for IbcEvent {
-    type Error = EventError;
-
-    fn try_from(
-        namada_event: &Event,
-    ) -> std::result::Result<Self, Self::Error> {
-        Ok(Self {
-            event_type: validate_ibc_event_type(namada_event)?,
-            #[allow(deprecated)]
-            attributes: namada_event
-                .attributes()
-                .iter()
-                .map(|(k, v)| (k.clone(), v.clone()))
-                .collect(),
-        })
-    }
-}
-
-impl TryFrom<Event> for IbcEvent {
-    type Error = EventError;
-
-    fn try_from(namada_event: Event) -> std::result::Result<Self, Self::Error> {
-        Ok(Self {
-            event_type: validate_ibc_event_type(&namada_event)?,
-            attributes: {
-                let mut attrs: HashMap<_, _> =
-                    namada_event.into_attributes().into_iter().collect();
-                attrs.with_attribute(event_domain_of::<Self>());
-                attrs
-            },
-        })
-    }
-}
-
```

Comment:
I think this could have still been useful, but not to the ledger codebase directly. it could be useful as sdk library code for users to consume. maybe this can be restored in the future, with proper event type validation